wordpress-mobile/WordPress-iOS
# Implement File Block - I

gutenberg PR: https://github.com/WordPress/gutenberg/pull/26278
gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2743

This is a first step on implementing **File Block**.

This implementation only considers the basics:
- Insert and render a File Block.
- Upload a file from the iOS native Documents Picker.
  - Don't let uploading file types not allowed by the blog.
- Edit the file name and Download button text.

The rest of features will be implemented in future PRs

**EDIT:** `PICK A FILE` has been changed to `CHOOSE A FILE`.
![file_01](https://user-images.githubusercontent.com/9772967/96686079-58b50f00-137e-11eb-9a8a-6b45a227a0b9.png)

To test:

- Add a File block.
- Upload a file from the device.
  - Check that you are able to upload images, videos, audio, pdf... but not allowed to upload txt files.
- Check that you can edit the file name and the Download button text.
- Check on HTML mode that the HREF is the remote one (not local).
- Check on preview that you can download the uploaded file.

PR submission checklist:
